x-pack/plugins/security_solution/public/cloud_security_posture/components/csp_details/insights_tab_csp.tsx

```diff
@@ -5,19 +5,119 @@
  * 2.0.
  */
 
-import React, { memo } from 'react';
-import { EuiSpacer } from '@elastic/eui';
+import React, { memo, useMemo, useState } from 'react';
+import type { EuiButtonGroupOptionProps } from '@elastic/eui';
+import { EuiButtonGroup, EuiSpacer } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import type { FlyoutPanelProps, PanelPath } from '@kbn/expandable-flyout';
+import { useExpandableFlyoutState } from '@kbn/expandable-flyout';
+import { i18n } from '@kbn/i18n';
+import type { FlyoutPanels } from '@kbn/expandable-flyout/src/store/state';
+import { CspInsightLeftPanelSubTab } from '../../../flyout/entity_details/shared/components/left_panel/left_panel_header';
 import { MisconfigurationFindingsDetailsTable } from './misconfiguration_findings_details_table';
+import { VulnerabilitiesFindingsDetailsTable } from './vulnerabilities_findings_details_table';
 
 /**
  * Insights view displayed in the document details expandable flyout left section
  */
+
+interface CspFlyoutPanelProps extends FlyoutPanelProps {
+  params: {
+    path: PanelPath;
+    hasMisconfigurationFindings: boolean;
+    hasVulnerabilitiesFindings: boolean;
+  };
+}
+interface CspFlyoutPanels extends FlyoutPanels {
+  left: CspFlyoutPanelProps;
+}
+
+// Type guard to check if the path is a PathPanel
+const isPathPanel = (path: PanelPath): path is PanelPath => {
+  return typeof path?.tab === 'string';
+};
+
 export const InsightsTabCsp = memo(
   ({ name, fieldName }: { name: string; fieldName: 'host.name' | 'user.name' }) => {
+    /* Using type cast here because originally I thought paths are part of Params when its actually its own field. The correct
+    way to solve this is by putting path outside params for every single openLeftPanel call in this folder, but given the time frame
+    We decided to do this for now */
+    const panels = useExpandableFlyoutState() as CspFlyoutPanels;
+
+    const getDefaultTab = () =>
+      panels.left?.params?.hasMisconfigurationFindings
```

Comment:
I would consider storing these values in variables as they are being reused a couple of times